app/routes.py

```diff
@@ -1,2 +1,200 @@
+                            "is_complete": False if task.completed_at is None else task.completed_at
+                        }}
+    if request.method == 'GET' and task.goal_id is None:
+        return one_task
+    elif request.method == 'GET':
+        one_task['task']['goal_id'] = task.goal_id
+        return one_task
+    elif request.method == 'PUT':
+        updates = request.get_json()
+        task.title = updates['title']
+        task.description = updates['description']
+        if 'completed_at' in updates.keys():
+            task.completed_at = updates['completed_at'] if not updates['completed_at'] \
+                                else datetime.utcnow()
+        
+        db.session.commit()
+        return make_response({"task": { "id": task.task_id,
+                                    "title": task.title,
+                                    "description": task.description,
+                                    "is_complete": False if task.completed_at is None else True
+                                    }})
+    elif request.method == 'DELETE':
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f"Task {task.task_id} \"{task.title}\" successfully deleted"})
+
+@tasks_bp.route('/<task_id>/mark_complete', methods=['PATCH'])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    if task is None: 
+        return jsonify(task), 404
+    else: 
+        task.completed_at = datetime.now()
+        url = 'https://slack.com/api/chat.postMessage'
+        channel_id = "C02J08B9S0N"
+        data = {
+                "channel": channel_id,
+                "token": SLACK_TOKEN,
+                "text": "Monica, is that you?? You hella forgot what you named your bot -_-"
+               }
+        r = requests.post(url, data=data)
+        
+    db.session.commit()
+    return make_response({"task": {"id": task.task_id,
+                                   "title": task.title,
+                                   "description": task.description,
+                                   "is_complete": True
+                                  }})
+
+@tasks_bp.route('/<task_id>/mark_incomplete', methods=['PATCH'])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return jsonify(task), 404
+    else:
+        task.completed_at = None
+    db.session.commit()
+    return make_response({"task": {"id": task.task_id,
+                                   "title": task.title,
+                                   "description": task.description,
+                                   "is_complete": False
+                                  }})
+
+@goals_bp.route('', methods=['POST', 'GET'])
+def handle_goals():
+    if request.method == 'GET':
+        goals = Goal.query.all()
+        goals_list = []
+        for goal in goals:
+            goals_list.append({"id": goal.goal_id,
+                               "title": goal.title})
```

Comment:
you could move this into a helper function or into your goal model as an instance method.